Source/MethodCall.cs

```diff
@@ -253,67 +245,44 @@ public void EvaluatedSuccessfully()
 
 		public virtual void Execute(ICallContext call)
 		{
-			this.Invoked = true;
-			this.CallCount++;
+			++this.callCount;
 
-			if (expectedMaxCallCount.HasValue && this.CallCount > expectedMaxCallCount)
+			if (expectedMaxCallCount.HasValue && this.callCount > expectedMaxCallCount)
 			{
 				if (expectedMaxCallCount == 1)
 				{
 					throw new MockException(
 						MockException.ExceptionReason.MoreThanOneCall,
-						Times.AtMostOnce().GetExceptionMessage(FailMessage, SetupExpression.ToStringFixed(), this.CallCount));
+						Times.AtMostOnce().GetExceptionMessage(this.failMessage, this.originalExpression.ToStringFixed(), this.callCount));
 				}
 				else
 				{
 					throw new MockException(
 						MockException.ExceptionReason.MoreThanNCalls,
-						Times.AtMost(expectedMaxCallCount.Value).GetExceptionMessage(FailMessage, SetupExpression.ToStringFixed(), this.CallCount));
+						Times.AtMost(expectedMaxCallCount.Value).GetExceptionMessage(this.failMessage, this.originalExpression.ToStringFixed(), this.callCount));
 				}
 			}
 
-			if (setupCallback != null)
-			{
-				setupCallback(call.Arguments);
-			}
+			this.callbackResponse?.Invoke(call.Arguments);
 
-			if (thrownException != null)
-			{
-				throw thrownException;
-			}
+			this.raiseEventResponse?.RespondTo(call);
 
-			if (this.mockEvent != null)
+			if (this.throwExceptionResponse != null)
 			{
-				if (mockEventArgsParams != null)
-				{
-					this.Mock.DoRaise(this.mockEvent, mockEventArgsParams);
-				}
-				else
-				{
-					var argsFuncType = mockEventArgsFunc.GetType();
-					if (argsFuncType.GetTypeInfo().IsGenericType && argsFuncType.GetGenericArguments().Length == 1)
-					{
-						this.Mock.DoRaise(this.mockEvent, (EventArgs)mockEventArgsFunc.InvokePreserveStack());
-					}
-					else
-					{
-						this.Mock.DoRaise(this.mockEvent, (EventArgs)mockEventArgsFunc.InvokePreserveStack(call.Arguments));
-					}
-				}
```

Comment:
This whole event triggering logic is moved into the `RaiseEventResponse` nested class. at the end of the file.